### DISCUSSION TITLE
Validating a dynamic field

### DISCUSSION THREAD

**Discussion by yarapolana** [adonisjs/core#1659] [Help]
I was looking for ways to authenticate a user with both email and username and came across this.

![EW7YV8SXYAAMkVJ](https://user-images.githubusercontent.com/19730118/94156043-25f72400-fe6f-11ea-89c1-7a0c65c92b54.jpg)

I was wondering if there is a possibility to validate this field?
Something similar to this:

```
  emailOrUsername: schema.string({}, [rules.email()] || schema.string({}) 
  // this field should either email or username but the validation falls to email
```

**Comment by thetutlage** [comment]
How do you validate username? Do you some regex for that or does any string is a valid username in your app?

**Reply by yarapolana** [reply]
my username is something like this      `schema.string({}, [rules.regex(/^[a-zA-Z0-9_-]{2,80}$/)])`

**Reply by thetutlage** [reply]
Okay. So yes, there is no way to define `OR` conditions when using rules. It is an intentional limitation, otherwise it makes composing a schema too complicated. A practical option is to define your own rule called `uid`, which performs the `OR` check internally.  Here's a guide on creating custom validator rules. https://preview.adonisjs.com/guides/validator/custom-rules.
